src/vs/workbench/contrib/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -1040,4 +1044,80 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 	private isExtensionAllowedToBeRecommended(id: string): boolean {
 		return this._allIgnoredRecommendations.indexOf(id.toLowerCase()) === -1;
 	}
-}
+
+	// {{SQL CARBON EDIT}}
+	promptRecommendedExtensionsByScenario(scenarioType: string): void {
+		const storageKey = 'extensionAssistant/RecommendationsIgnore/' + scenarioType;
+
+		if (this.storageService.getBoolean(storageKey, StorageScope.GLOBAL, false)) {
+			return;
+		}
+
+		let recommendations: IExtensionRecommendation[];
+		let localExtensions: ILocalExtension[];
+		const getRecommendationPromise = this.getRecommendedExtensionsByScenario(scenarioType).then(recs => { recommendations = recs; });
+		const getLocalExtensionPromise = this.extensionsService.getInstalled(ExtensionType.User).then(local => { localExtensions = local; });
+		Promise.all([getRecommendationPromise, getLocalExtensionPromise]).then(() => {
+			if (!recommendations.every(rec => { return localExtensions.findIndex(local => local.identifier.id.toLocaleLowerCase() === rec.extensionId.toLocaleLowerCase()) !== -1; })) {
+				return new Promise<void>(c => {
+					this.notificationService.prompt(
+						Severity.Info,
+						localize('ExtensionsRecommended', "Azure Data Studio has extension recommendations."),
+						[{
+							label: localize('installAll', "Install All"),
+							run: () => {
+								this.telemetryService.publicLog(scenarioType + 'Recommendations:popup', { userReaction: 'install' });
+								const installAllAction = this.instantiationService.createInstance(InstallRecommendedExtensionsByScenarioAction, scenarioType, recommendations);
+								installAllAction.run();
+								installAllAction.dispose();
+
+								if (scenarioType === LocalizedConstants.visualizerExtensions) {
```

Comment:
I think I left a comment for this. I don’t think we should have this logic here. please check scenario and show notification logic.